processing/src/test/java/io/druid/query/search/SearchQueryRunnerTest.java

```diff
@@ -455,41 +562,52 @@ public void testSearchNonExistingDimension()
     );
   }
 
-  private void checkSearchQuery(SearchQuery searchQuery, Map<String, Set<String>> expectedResults)
+  private void checkSearchQuery(Query searchQuery, Map<String, Set<String>> expectedResults)
+  {
+    List<SearchHit> expectedHits = Lists.newLinkedList();
+    for (Map.Entry<String, Set<String>> entry : expectedResults.entrySet()) {
+      String dimension = entry.getKey();
+      for (String value : entry.getValue()) {
+        expectedHits.add(new SearchHit(dimension, value));
+      }
+    }
+    checkSearchQuery(searchQuery, runner, expectedHits);
+  }
+
+  private void checkSearchQuery(Query searchQuery, QueryRunner runner, List<SearchHit> expectedResults)
   {
-    HashMap<String,List> context = new HashMap<String, List>();
     Iterable<Result<SearchResultValue>> results = Sequences.toList(
-        runner.run(searchQuery, context),
+        runner.run(searchQuery, ImmutableMap.of()),
         Lists.<Result<SearchResultValue>>newArrayList()
     );
+    checkSearchQuery(expectedResults, results);
+  }
 
+  private void checkSearchQuery(
+      List<SearchHit> expectedResults,
+      Iterable<Result<SearchResultValue>> results
+  )
+  {
     for (Result<SearchResultValue> result : results) {
       Assert.assertEquals(new DateTime("2011-01-12T00:00:00.000Z"), result.getTimestamp());
       Assert.assertTrue(result.getValue() instanceof Iterable);
 
       Iterable<SearchHit> resultValues = result.getValue();
       for (SearchHit resultValue : resultValues) {
-        String dimension = resultValue.getDimension();
-        String theValue = resultValue.getValue();
-        Assert.assertTrue(
-            String.format("Result had unknown dimension[%s]", dimension),
-            expectedResults.containsKey(dimension)
-        );
-
-        Set<String> expectedSet = expectedResults.get(dimension);
-        Assert.assertTrue(
-            String.format("Couldn't remove dim[%s], value[%s]", dimension, theValue), expectedSet.remove(theValue)
-        );
+        int index = expectedResults.indexOf(resultValue);
+        if (index < 0) {
+          Assert.fail("Not expected result " + resultValue);
```

Comment:
This makes it sound like the result was incorrect, but I think in this case no result was found at all for the dimension and value, is that right?  If so, maybe it would be better to say something like "No result found containing " + resultValue.getDimension() + " and " resultValue.getValue().  (That seems to make sense whether we were expecting a cardinality to be in it or not.)

it might even be helpful, then, to print all of the results that were present, unless that might be too much output.  I don't think that would be the case at this point, though of course nothing prevents the test changing in the future.